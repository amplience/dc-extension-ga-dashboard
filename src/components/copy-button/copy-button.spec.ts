import { render } from '@testing-library/svelte';
import CopyButton from './copy-button.svelte';
import { tick } from 'svelte';

describe('CopyButton', () => {
  const mockWriteText = jest.fn();
  const mockReadText = jest.fn();

  beforeEach(() => {
    Object.assign(navigator, {
      clipboard: {
        writeText: mockWriteText,
        readText: mockReadText,
      },
    });
  });

  it('should render a copy button', async () => {
    const { container } = render(CopyButton, { value: 'test', title: 'title' });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add some text to the clipboard when clicked', async () => {
    mockWriteText.mockResolvedValue(true);
    mockReadText.mockResolvedValue('test');

    const { container } = render(CopyButton, { value: 'test' });
    expect(container.firstChild).toMatchSnapshot();

    const buttonElement = document.querySelector('button');
    (buttonElement as HTMLElement).click();

    await tick();
    await expect(navigator.clipboard.readText()).resolves.toEqual('test');
  });
});
