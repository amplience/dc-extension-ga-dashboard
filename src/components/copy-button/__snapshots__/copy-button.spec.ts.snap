// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyButton should add some text to the clipboard when clicked 1`] = `
<div>
  <button
    aria-label="Copy"
    class="svelte-23tqn3"
    title=""
  >
    <object
      src="ic-copyicon.svg/"
    />
  </button>
</div>
`;

exports[`CopyButton should render a copy button 1`] = `
<div>
  <button
    aria-label="title"
    class="svelte-23tqn3"
    title="title"
  >
    <object
      src="ic-copyicon.svg/"
    />
  </button>
</div>
`;
